### PR TITLE
[#35] [Chore] UI Integration Test - Sign In page

### LINKED ISSUE #35: [Chore] UI Integration Test - Sign In page
### Why
Add UI integration testing to make sure the app's flow works as expected.

## Acceptance Criteria
- Add the test cases for the Sign In page:
  - When inputting an invalid email or password, it shows the error message below each text field.
  - When inputting a valid email and password and signing in successfully, it returns the Dashboard page.
  - When inputting a valid email and password and signing in unsuccessfully, it shows an error message.

### PR DESCRIPTION
**Pull request by manh-t**
Close #35 

## What happened 👀

- Add the fake response json files for:
  - POST `api/v1/oauth/token`
  - GET `api/v1/surveys`
  - GET `api/v1/me`
- Add the test cases for SignIn screen:
  - When the user submits the valid submission, he should be navigated to Dashboard screen
  - When the user submits the invalid submission, he should see the Alert dialog.
  - When the user submits the submission but failed on the validation step, he should see the error popup from the browser.

## Insight 📝

N/A

## Proof Of Work 📹

image

### CODE REVIEW

**Comment by manh-t** [comment]
🚀 Deployed on https://64ec76e213ef6f02c5f4ba7e--manh-react-survey.netlify.app

**Comment by manh-t** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Congrats, coverage is good</td>
    </tr>
  </tbody>
</table>

## Coverage in All Files
File | Line Coverage | Statement Coverage | Function Coverage | Branch Coverage
---- | ------------: | -----------------: | ----------------: | --------------:
[src/screens/Survey/index.tsx](../blob/748ba21/src/screens/Survey/index.tsx) | (1/3) 33% | (1/3) 33% | (0/1) 0% | (0/0) NaN%
[src/tests/error.ts](../blob/748ba21/src/tests/error.ts) | (2/3) 67% | (2/3) 67% | (1/2) 50% | (1/3) 33%
[src/../Content/index.tsx](../blob/748ba21/src/components/Dashboard/Content/index.tsx) | (12/15) 80% | (12/15) 80% | (5/7) 71% | (5/9) 56%
[src/../Dashboard/index.tsx](../blob/748ba21/src/screens/Dashboard/index.tsx) | (10/12) 83% | (10/12) 83% | (5/7) 71% | (9/10) 90%
[src/App.tsx](../blob/748ba21/src/App.tsx) | (3/3) 100% | (3/3) 100% | (1/1) 100% | (0/0) NaN%
[src/index.tsx](../blob/748ba21/src/index.tsx) | (3/3) 100% | (3/3) 100% | (0/0) NaN% | (0/0) NaN%
[src/../Authentication/index.ts](../blob/748ba21/src/adapters/Authentication/index.ts) | (2/2) 100% | (2/2) 100% | (1/1) 100% | (0/0) NaN%
[src/adapters/Base/index.ts](../blob/748ba21/src/adapters/Base/index.ts) | (8/8) 100% | (8/8) 100% | (2/2) 100% | (2/2) 100%
[src/adapters/BaseAuth/index.ts](../blob/748ba21/src/adapters/BaseAuth/index.ts) | (10/10) 100% | (10/10) 100% | (4/4) 100% | (2/2) 100%
[src/adapters/Survey/index.ts](../blob/748ba21/src/adapters/Survey/index.ts) | (1/1) 100% | (1/1) 100% | (1/1) 100% | (0/0) NaN%
[src/adapters/User/index.ts](../blob/748ba21/src/adapters/User/index.ts) | (1/1) 100% | (1/1) 100% | (1/1) 100% | (0/0) NaN%
[src/components/Alert/index.tsx](../blob/748ba21/src/components/Alert/index.tsx) | (3/3) 100% | (3/3) 100% | (2/2) 100% | (0/0) NaN%
[src/../index.tsx](../blob/748ba21/src/components/BackgroundImage/index.tsx) | (3/3) 100% | (3/3) 100% | (1/1) 100% | (2/2) 100%
[src/../Empty/index.tsx](../blob/748ba21/src/components/Dashboard/Empty/index.tsx) | (3/3) 100% | (3/3) 100% | (1/1) 100% | (0/0) NaN%
[src/../Header/index.tsx](../blob/748ba21/src/components/Dashboard/Header/index.tsx) | (3/3) 100% | (3/3) 100% | (1/1) 100% | (9/9) 100%
Other (28 more) | (170/170) 100% | (170/170) 100% | (60/60) 100% | (39/41) 95%
Total | (235/243) 97% | (235/243) 97% | (86/92) 93% | (69/78) 88%

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 11c3ca8bd3c3add3b2de52a6b7c4290e4d744e46
</p>

File: cypress/e2e/SignIn/index.spec.ts [hunk 1]
```diff
@@ -1,10 +1,76 @@
...
+  forgotButton: 'sign-in-form__forgot-button',
+  signInButton: 'sign-in-form__button',
+  errorAlert: 'sign-in__error-alert',
+  loadingDialog: 'sign-in__loading-dialog',
+};
+
+describe('SignIn screen', () => {
+  context('given the valid form submission', () => {
+    beforeEach(() => {
+      cy.interceptWithDelay('GET', '**/api/v1/surveys*', 200, {
```
Comment (tyrro) [inline cypress/e2e/SignIn/index.spec.ts]: Should we move `200` into a constant?

File: cypress/e2e/SignIn/index.spec.ts [hunk 2]
```diff
@@ -1,10 +1,76 @@
...
+  passwordLabel: 'sign-in-form__password',
+  passwordField: 'sign-in-form__input-password',
+  forgotButton: 'sign-in-form__forgot-button',
+  signInButton: 'sign-in-form__button',
+  errorAlert: 'sign-in__error-alert',
+  loadingDialog: 'sign-in__loading-dialog',
+};
+
+describe('SignIn screen', () => {
+  context('given the valid form submission', () => {
```
Comment (tyrro) [inline cypress/e2e/SignIn/index.spec.ts]:
```suggestion
  context('given a valid form submission', () => {
```

File: cypress/e2e/SignIn/index.spec.ts [hunk 3]
```diff
@@ -1,10 +1,76 @@
...
+    beforeEach(() => {
+      cy.interceptWithDelay('GET', '**/api/v1/surveys*', 200, {
+        statusCode: 200,
+        fixture: 'Survey/List/valid.json',
+      });
+      cy.interceptWithDelay('GET', '**/api/v1/me', 200, {
```
Comment (tyrro) [inline cypress/e2e/SignIn/index.spec.ts]:
```suggestion

      cy.interceptWithDelay('GET', '**/api/v1/me', 200, {
```

**Review by tyrro — APPROVED** [review]

File: cypress/e2e/SignIn/index.spec.ts (hunk 1 above)
Comment (manh-t) [reply cypress/e2e/SignIn/index.spec.ts]: @tyrro updated in [63408c1](https://github.com/manh-t/react-survey/pull/62/commits/63408c179a51976aa711afd78e2eddc1e76492f2)

File: cypress/e2e/SignIn/index.spec.ts [hunk 4]
```diff
@@ -1,10 +1,81 @@
...
+      });
+
+      cy.visit('/');
+      cy.findByTestId(signInScreenTestIds.emailField).type('test@email.com');
+      cy.findByTestId(signInScreenTestIds.passwordField).type('12345678');
+      cy.findByTestId(signInScreenTestIds.signInButton).click();
+
+      cy.findByTestId(signInScreenTestIds.loadingDialog).should('be.visible');
+
+      cy.findByTestId(signInScreenTestIds.errorAlert).should('be.visible');
```
Comment (nvminhtue) [inline cypress/e2e/SignIn/index.spec.ts:67]: how about showing the error msg as well?

File: package-lock.json
Comment (nvminhtue) [inline package-lock.json:1]: there's no change in package.json but lock, does it belong to the previous PR? Please update it to the corresponding changes

**Review by nvminhtue — APPROVED** [review]

File: cypress/e2e/SignIn/index.spec.ts (hunk 4 above)
Comment (manh-t) [reply cypress/e2e/SignIn/index.spec.ts:67]: @nvminhtue thanks, I added the error message's validation 😄

**Pull request merged**